packages/phpstan-rules/src/Rules/Explicit/SameNamedParamFamilyRule.php

```diff
@@ -0,0 +1,200 @@
+        if (! $phpMethodReflection instanceof PhpMethodReflection) {
+            return [];
+        }
+
+        $currentParamNames = $this->resolveClassMethodParamNames($node);
+        $parentParamNames = $this->resolveMethodReflectionParamNames($phpMethodReflection);
+
+        $conflictingParamNames = $this->resolveConflictingParamNames($currentParamNames, $parentParamNames);
+
+        // everything is the same
+        if ($conflictingParamNames === []) {
+            return [];
+        }
+
+        if (! $this->isContainerBuilderMissmatch($conflictingParamNames, $phpMethodReflection)) {
+            return [self::ERROR_MESSAGE];
+        }
+        return [];
+    }
+
+    /**
+     * @param string[] $conflictingParamNames
+     */
+    private function isContainerBuilderMissmatch(
+        array $conflictingParamNames,
+        PhpMethodReflection $phpMethodReflection
+    ): bool
+    {
+        if ($conflictingParamNames !== ['containerBuilder']) {
+            return false;
+        }
+
+        $parentClassReflection = $phpMethodReflection->getDeclaringClass();
+        // the Container vs ContainerBuilder missmatch
+        return in_array($parentClassReflection->getName(), [
+            ExtensionInterface::class,
+            ConfigurationExtensionInterface::class,
+            Bundle::class,
+            BundleInterface::class,
+            Kernel::class,
+            CompilerPassInterface::class,
+        ], true);
+    }
+
+    /**
+     * @return string[]
+     */
+    private function resolveMethodReflectionParamNames(PhpMethodReflection $phpMethodReflection): array
+    {
+        $paramNames = [];
+
+        $firstVariant = $phpMethodReflection->getVariants()[0];
+        foreach ($firstVariant->getParameters() as $parameterReflectionWithPhpDoc) {
+            $paramNames[] = $parameterReflectionWithPhpDoc->getName();
+        }
+
+        return $paramNames;
+    }
+
+    /**
+     * @return string[]
+     */
+    private function resolveClassMethodParamNames(ClassMethod $classMethod): array
+    {
+        $paramNames = [];
+        foreach ($classMethod->params as $param) {
+            /** @var string $paramName */
+            $paramName = $this->simpleNameResolver->getName($param->var);
+            $paramNames[] = $paramName;
+        }
+
+        return $paramNames;
+    }
+
+    private function shouldSkip(Scope $scope, ClassMethod $classMethod): bool
+    {
+        $classReflection = $scope->getClassReflection();
+        if (! $classReflection instanceof ClassReflection) {
+            return true;
+        }
+
+        if ($classMethod->params === []) {
+            return true;
+        }
+
+        // no parent interface, class nor trait
+        return count($classReflection->getAncestors()) === 1;
+    }
+
+    /**
+     * @param string[] $currentParamNames
+     * @param string[] $parentParamNames
+     * @return string[]
```

Comment:
To enforce the newly indexed return-array

```suggestion
     * @return list<string>
```

Reply:
I have never used this syntax. This would have to be standardized in CI somehow.